From 1dea0b11dd7cbdc963f5122e89b10699a0564220 Mon Sep 17 00:00:00 2001
From: Tydus <noreply@example.com>
Date: Thu, 20 Feb 2020 19:57:11 +0900
Subject: [PATCH] rtmp-module: Don't fail if no stream name specified

---
 ngx_rtmp_cmd_module.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ngx_rtmp_cmd_module.c b/ngx_rtmp_cmd_module.c
index e7826aa..14c1fdb 100644
--- a/ngx_rtmp_cmd_module.c
+++ b/ngx_rtmp_cmd_module.c
@@ -553,7 +553,7 @@ ngx_rtmp_cmd_publish_init(ngx_rtmp_session_t *s, ngx_rtmp_header_t *h,
         ngx_log_error(NGX_LOG_ERR, s->connection->log, 0,
                       "publish: no stream name specified");
 
-        return NGX_ERROR;
+        //return NGX_ERROR;
     }
 
     if (ngx_rtmp_process_request_line(s, v.name, v.args,
-- 
2.7.4

